### DISCUSSION TITLE
Windows installation

### DISCUSSION THREAD

**Discussion by FGarelli** [OpenRadioss/OpenRadioss#1449] [Q&A] [answered]
Hi, I'm trying to install Openradioss but I have this error when I start to install the starter pack:

![image](https://user-images.githubusercontent.com/132582318/236837360-e5038076-657b-409b-b808-39c05768688f.png)

What should i do?

**Comment by MarianBulla** [comment]
Hi,
can you try to use a folder without special characters, like " " (space), please ?
Something like "C:/tmp/openradioss/".

Maybe it's due to these "issues" with path names in windows.
I'm using: "D:\WS\GitHub\OpenRadioss" and it works fine.

Some questions to circle down a possible other sources of this issue, if the above will not work:
- Which windows version are you using ?
- When did you install the Intel OneAPI ?

Hope it helps you ?

Thanks and best
Marian

**Reply by FGarelli** [reply]
I moved to a folder without the spaces and it didn't work.
I'm using windows 10.
I installed Intel OneAPI a week ago.

**Comment by MarianBulla** [comment]
Yes, I was able to reproduce it.
With Visual Studio 2022, it works, well.
Anyway, it should also from command window.
Will have a deeper look at it.

Thank you very much for pointing to it.

Thanks and best regards,
Marian

**Reply by FGarelli** [reply]
Is anything more that I can try? I tried to use visual studio 2022, the command window and cygwin. Any suggestion?

**Reply by MarianBulla** [reply]
Yes, 
did you lounched the configuration script from the Intel OneAPI compiler ?

call "c:\Program Files (x86)\Intel\oneAPI\setvars.bat"

If I do it in my cmd window, then it works fine with the compilation.

If I don't do it, I'm getting the same error message, you are showing.

Hope it helps you ?

Thanks and best reagrds,
Marian

![image](https://user-images.githubusercontent.com/105412502/237024829-ddc10fd2-2643-4203-a4d8-5a4b31b47a84.png)

**Reply by FGarelli** [reply]
Yes, i did the same and i have this error:

<img width="568" alt="image" src="https://user-images.githubusercontent.com/132582318/237027225-a1d35001-ba99-47cc-ba6c-7d7ae8b520ef.png">
<img width="482" alt="image" src="https://user-images.githubusercontent.com/132582318/237027325-3937dd92-1dca-4d0e-bceb-335c8d072f92.png">

Is it possible that i missed some intallation packages?

**Reply by FGarelli** [reply]
on visual studio i have this error
<img width="953" alt="image" src="https://user-images.githubusercontent.com/132582318/237036088-e8e16af5-619f-4b8e-bfc4-2e5733a9aac8.png">

**Reply by MarianBulla** [reply]
It looks like your Visual Studio installation is only the 32bit version and causes the Intel OneAPI some troubles.
The windows version of OpenRadioss needs some of the VS components to be able to run the compiler.
Can you try to uninstall the VS and re-install the 64bit version, please ?
Maybe a new re-installation of the Intel OneAPI will be required.
Then you can try (in the cmd window) the following:
>ifx
>icx /v

You should get something like this from the compiler:
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/f684a636-0563-4190-9928-158e52ecd9b7)


Hope it helps you ?

Thanks and best 
Marian

**Reply by FGarelli** [reply]
I missed some components from the Intel API HPC Toolkit. Now i manage to install the starter and the engine!

**Reply by MarianBulla** [reply, marked as answer]
Great !
Congratulations !!!

Have fun with the solver, now.

Thanks and best regards,
Marian

**Comment by roderaga** [comment]
Hi,

Thank you for the work with the open source. 

Does someone has the same problem that after installating the source software by downloading it and installing via Intel oneApli Tools with the corresponsing .exes(starter and engine) and submitting succesfully one example from the website, next day opening the laptop and laucnhing the exe files again wit hthe same input files, it has the following FAILED.

![failed](https://github.com/OpenRadioss/OpenRadioss/assets/105799620/ee05e9f4-47ff-4957-a1a2-2db63817432e)

Thank you,
Oscar

**Reply by MarianBulla** [reply]
Hi Oscar,

thanks for your interest in OpenRadioss. 
Also thanks for your screenshot.
It helps a lot of identifying, what's going wrong :-) 
In your case, it seems, you are recompiling OpenRadioss....which went successful till the end. But while the script tried to copy the fresh backed executable back to the .../exec/ - folder, windows prevented it.

One reason could be: you're already using the "engine_win64.exe"
Please check your windows task-manager, if there is such a process, which is still running.
If it is so, then stop/end/kill this process.

Another reason could be, you are compiling in an other command window.  --> End and close it.

If nothing helps, you may use the very old windwos trick: --> Reboot your windows machine and try again :-) 

Hope it helps you ?

Thanks and best regards,
Marian

**Reply by roderaga** [reply]
I see,

This could be also due to some vpn secutiry problem?

Thank you very much, Marian.

Best Regards,
Oscar

**Comment by cunyizju** [comment]
Hi, I am using Win10+OneAPI2024+VS2022 and encountering the same error. <img alt="image" width="953" src="https://user-images.githubusercontent.com/132582318/237036088-e8e16af5-619f-4b8e-bfc4-2e5733a9aac8.png">


For HPC, I have installed the all packages. Do you have any ideas to fix this error? Thanks a lot!

**Reply by cunyizju** [reply]
**details about CMake error:** 

Severity	Code	Description	Project	File	Line	Suppression State	Details
Error		CMake Error at D:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.27/Modules/CMakeTestCCompiler.cmake:67 (message):
  The C compiler

    "D:/Program Files (x86)/Intel/oneAPI/compiler/2024.0/bin/icx.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: 'E:/A_RepoEternal/OpenRadioss/vs_build_win64/Starter_x64-Release/CMakeFiles/CMakeScratch/TryCompile-zd7cap'
    
    Run Build Command(s): "D:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v cmTC_452dd
    [1/2] "D:\Program Files (x86)\Intel\oneAPI\compiler\2024.0\bin\icx.exe"  /nologo   /DWIN32 /D_WINDOWS /W3  /MDd /Zi /Ob0 /Od /RTC1 -QMD -QMT CMakeFiles\cmTC_452dd.dir\testCCompiler.c.obj -QMF CMakeFiles\cmTC_452dd.dir\testCCompiler.c.obj.d /FoCMakeFiles\cmTC_452dd.dir\testCCompiler.c.obj /FdCMakeFiles\cmTC_452dd.dir\ -c E:\A_RepoEternal\OpenRadioss\vs_build_win64\Starter_x64-Release\CMakeFiles\CMakeScratch\TryCompile-zd7cap\testCCompiler.c
    [2/2] cmd.exe /C "cd . && "D:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E vs_link_exe --intdir=CMakeFiles\cmTC_452dd.dir --rc="D:\Windows Kits\10\bin\10.0.22621.0\x64\rc.exe" --mt="D:\Windows Kits\10\bin\10.0.22621.0\x64\mt.exe" --manifests  -- "D:\Program Files (x86)\Intel\oneAPI\compiler\2024.0\bin\icx.exe" /nologo CMakeFiles\cmTC_452dd.dir\testCCompiler.c.obj  /Qoption,link,/machine:x64  /debug /INCREMENTAL /Qoption,link,/subsystem:console  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /link /out:cmTC_452dd.exe /implib:cmTC_452dd.lib /pdb:cmTC_452dd.pdb /version:0.0  && cd ."
    FAILED: cmTC_452dd.exe 
    cmd.exe /C "cd . && "D:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E vs_link_exe --intdir=CMakeFiles\cmTC_452dd.dir --rc="D:\Windows Kits\10\bin\10.0.22621.0\x64\rc.exe" --mt="D:\Windows Kits\10\bin\10.0.22621.0\x64\mt.exe" --manifests  -- "D:\Program Files (x86)\Intel\oneAPI\compiler\2024.0\bin\icx.exe" /nologo CMakeFiles\cmTC_452dd.dir\testCCompiler.c.obj  /Qoption,link,/machine:x64  /debug /INCREMENTAL /Qoption,link,/subsystem:console  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /link /out:cmTC_452dd.exe /implib:cmTC_452dd.lib /pdb:cmTC_452dd.pdb /version:0.0  && cd ."
    LINK Pass 1: command "D:\Program Files (x86)\Intel\oneAPI\compiler\2024.0\bin\icx.exe /nologo CMakeFiles\cmTC_452dd.dir\testCCompiler.c.obj /Qoption,link,/machine:x64 /debug /INCREMENTAL /Qoption,link,/subsystem:console kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /link /out:cmTC_452dd.exe /implib:cmTC_452dd.lib /pdb:cmTC_452dd.pdb /version:0.0 /MANIFEST /MANIFESTFILE:CMakeFiles\cmTC_452dd.dir/intermediate.manifest CMakeFiles\cmTC_452dd.dir/manifest.res" failed (exit code 1104) with the following output:
    icx: remark: Note that use of '-debug' without any optimization-level option will turn off most compiler optimizations similar to use of '/Od'; use '-Rno-debug-disables-optimization' to disable this remark [-Rdebug-disables-optimization]

    LINK : fatal error LNK1104: cannot open file 'libircmt.lib'

    icx: error: linker command failed with exit code 1104 (use -v to see invocation)

    ninja: build stopped: subcommand failed.
    
    

  

  CMake will not be able to correctly generate this project.		D:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.27/Modules/CMakeTestCCompiler.cmake	67

**Reply by MarianBulla** [reply]
Hello cunyizju,
thanks for your interest on OpenRadioss, and sorry for the issue you face.

This is because of some changes done by INTEL in their new oneAPI compiler 2024.

Please uninstall it and install the 2023.2.0 version of the Intel oneAPI (and HPC) package.
This one should work. 

Hope it helps you ?

Thanks and best regards,
Marian

**Reply by cunyizju** [reply]
Dear Marian,

Many thanks for your timely reply. 
I will try the 2023.2.0 version.
Is there any plan to update the current code to be compatible with oneAPI compiler 2024?

Warm regards,
Cunyi

**Reply by cunyizju** [reply]
One more question: will OneAPI 2021 work? Intel does not provide 2023.2.0 version currently. They only provide the latest version.

**Reply by MacNeacail** [reply]
If you download the program, and use the cmake with VS 2022 and the latest oneapi you will get only two errors. Both appear trivial.  I am not going to waste time fixing a path command. 
The oneapi creates a directory called currentbuild where whatever version you are using will appear and it updates at this location.  The chances that the current build will not run this program are close to zero. 
However, the owners of this program should provide a sln for VS and get away from the cmake.  

https://learn.microsoft.com/en-us/cpp/build/cmake-projects-in-visual-studio?view=msvc-170

This link provides a step by step guide.  You are better off spending a few hours using the latest vs and compiler.

**Reply by MarianBulla** [reply]
Dear @cunyizju,
we did NOT test it with OneAPI 2021, so I'm not sure if it will work with this version.

Anyway, I would recomend you to use OneAPR 2023.2.0, since it is tested with that version. 

We are working with Intel to test the recent version.

Thanks and best regards,
Marian

**Comment by mehdioulal** [comment]
Hi all,
Thanks for the work with OpenRadioss. 
I got this error when I tried to build OpenRadioss Engine with Intel MPI support:
![image](https://github.com/OpenRadioss/OpenRadioss/assets/158052983/669cf907-c477-4282-9983-8826bce86982)
When trying to go to the folder directly, I only have a vars.bat file (see below). 
![image](https://github.com/OpenRadioss/OpenRadioss/assets/158052983/6a714ca6-44a0-4650-b68e-14e50aa96330)
I am using the 2024.0.1.45_offline OneAPI Kit with the 2024.0.1.35_offline HPC Kit with Windows (from [there](https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit-download.html))

Could you help me please? 
Thanks!

**Reply by OlivierWienholtz** [reply]
Hello,

It seams IntelMPI libraries were not found.
The PATH goes into this directory, but then goes 1 directory upper, and then lib/release/impi.lib

Hope it helps,

**Comment by JiancaiZheng** [comment]
Hi, i am using the "2024.2.1.101_online OneAPI Kit with the 2024.2.1.80_offline HPC Kit with Windows 11". In the "Intel oneAPI command prompt for Intel 64 for Visual Studio 2022" command, I got this error when I tried to build OpenRadioss Starter and Engine :
<img width="1027" alt="微信图片_20240909081815" src="https://github.com/user-attachments/assets/1fd04efe-b058-4c50-932e-c6bba8bf127f">

Could you help me please?
Thanks!

**Reply by OlivierWienholtz** [reply]
Hello,

It looks like Python is missing.
You need installed and available Python distribution to build OpenRadioss.
Pls check python is installed and configured.

Best Regards,
Olivier W.

**Reply by JiancaiZheng** [reply]
Dear Olivier,
Now i install the starter and the engine solver in Windows!

Thanks and best regards,
Jiancai Zheng

**Reply by OlivierWienholtz** [reply]
Great to see that you have been able to build Starter & Engine ! 

Best Regards,
Olivier
